### ISSUE TITLE
Is it possible to use open3d visualization using tensorboard on a headless server?

### ISSUE THREAD

**Issue by pycoco** [isl-org/Open3D#6689]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](https://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](https://www.open3d.org/docs/release/) and the [latest documentation](https://www.open3d.org/docs/latest/) (for `main` branch).

### My Question

I already build open3d from source and save point cloud. but i met below error:
`[Open3D INFO] WebRTCWindowSystem: HTTP handshake server disabled.
[Open3D INFO] WebRTC GUI backend enabled.
[Open3D INFO] Resetting default logger to print to terminal.
[Open3D INFO] Window window_0 created.
[Open3D INFO] EGL headless mode enabled.
[Open3D INFO] Set WEBRTC_STUN_SERVER environment variable add a customized WebRTC STUN server.
[Open3D INFO] WebRTC Jupyter handshake mode enabled.
FEngine (64 bits) created at 0x7f647c00c510 (threading is enabled)
eglInitialize failed
Fatal Python error: Segmentation fault`

**Comment by ssheorey** [comment]
Yes. Check that you have a GPU (integrated / discrete) on the headless server and that drivers are properly installed. `eglinfo` command should report a working EGL installation.

**Comment by pycoco** [comment]
> eglinfo

Thanks for your reply.  I don't known why eglInitialize failed.
EGL client extensions string:
    EGL_EXT_client_extensions EGL_EXT_device_base
    EGL_EXT_device_enumeration EGL_EXT_device_query EGL_EXT_platform_base
    EGL_KHR_client_get_all_proc_addresses EGL_KHR_debug
    EGL_EXT_platform_x11 EGL_MESA_platform_surfaceless
X11 platform:
eglinfo: eglInitialize failed

I  already followed the document to build open3d from source[master vertion]several times.  The command that i used:
cmake -DENABLE_HEADLESS_RENDERING=ON \
                 -DBUILD_GUI=OFF \
                 -DBUILD_WEBRTC=OFF \
                 -DUSE_SYSTEM_GLEW=OFF \
                 -DUSE_SYSTEM_GLFW=OFF \
                 ..
My opengl version is 3.3, should i upgrade opengl to 4.x?

**Comment by pycoco** [comment]
After i installed the whl file and run "python render_to_image.py". I met the error:
ImportError: cannot import name 'gui' from 'open3d.visualization'. Maybe it's caused by "-DBUILD_GUI=OFF"? But when i set -DENABLE_HEADLESS_RENDERING=ON, -DBUILD_GUI must be OFF.

**Comment by pycoco** [comment]
> Yes. Check that you have a GPU (integrated / discrete) on the headless server and that drivers are properly installed. `eglinfo` command should report a working EGL installation.

Any suggestions? I found that tensorboard plugin depends on O3DVisualizer, which in turn depends on GUI. So we can't use open3d visualization using tensorboard on a headless server.
